test/integration/podgroup_controller_test.go

```diff
@@ -0,0 +1,360 @@
+					midPriority).Label(v1alpha1.PodGroupLabel, "pg1-1").Node(nodeName).Phase(v1.PodRunning).Obj(),
+				st.MakePod().Namespace(ns).Name("t1-p1-3").Req(map[v1.ResourceName]string{v1.ResourceMemory: "50"}).Priority(
+					midPriority).Label(v1alpha1.PodGroupLabel, "pg1-1").Node(nodeName).Phase(v1.PodRunning).Obj(),
+			},
+			status: []*v1alpha1.PodGroup{
+				util.UpdatePGStatus(util.MakePG("pg1-1", ns, 3, nil, nil), "Running", "", 0, 3, 0, 0),
+			},
+			incomingPods: []*v1.Pod{
+				st.MakePod().Namespace(ns).Name("t1-p1-1").Req(map[v1.ResourceName]string{v1.ResourceMemory: "50"}).Priority(
+					midPriority).Label(v1alpha1.PodGroupLabel, "pg1-1").Node(nodeName).Phase(v1.PodSucceeded).Obj(),
+			},
+			want: []*v1alpha1.PodGroup{
+				util.UpdatePGStatus(util.MakePG("pg1-1", ns, 3, nil, nil), "Running", "", 0, 2, 1, 0),
+			},
+		},
+		{
+			name: "The status of the pod changes from running to failed",
+			podGroups: []*v1alpha1.PodGroup{
+				util.MakePG("pg1-1", ns, 3, nil, nil),
+			},
+			existingPods: []*v1.Pod{
+				st.MakePod().Namespace(ns).Name("t1-p1-1").Req(map[v1.ResourceName]string{v1.ResourceMemory: "50"}).Priority(
+					midPriority).Label(v1alpha1.PodGroupLabel, "pg1-1").Node(nodeName).Phase(v1.PodRunning).Obj(),
+				st.MakePod().Namespace(ns).Name("t1-p1-2").Req(map[v1.ResourceName]string{v1.ResourceMemory: "50"}).Priority(
+					midPriority).Label(v1alpha1.PodGroupLabel, "pg1-1").Node(nodeName).Phase(v1.PodRunning).Obj(),
+				st.MakePod().Namespace(ns).Name("t1-p1-3").Req(map[v1.ResourceName]string{v1.ResourceMemory: "50"}).Priority(
+					midPriority).Label(v1alpha1.PodGroupLabel, "pg1-1").Node(nodeName).Phase(v1.PodRunning).Obj(),
+			},
+			status: []*v1alpha1.PodGroup{
+				util.UpdatePGStatus(util.MakePG("pg1-1", ns, 3, nil, nil), "Running", "", 0, 3, 0, 0),
+			},
+			incomingPods: []*v1.Pod{
+				st.MakePod().Namespace(ns).Name("t1-p1-1").Req(map[v1.ResourceName]string{v1.ResourceMemory: "50"}).Priority(
+					midPriority).Label(v1alpha1.PodGroupLabel, "pg1-1").Node(nodeName).Phase(v1.PodFailed).Obj(),
+			},
+			want: []*v1alpha1.PodGroup{
+				util.UpdatePGStatus(util.MakePG("pg1-1", ns, 3, nil, nil), "Failed", "", 0, 2, 0, 1),
+			},
+		},
+		{
+			name: "The status of no podGroup label pod changes from pending to running",
+			podGroups: []*v1alpha1.PodGroup{
+				util.MakePG("pg1-1", ns, 3, nil, nil),
+			},
+			existingPods: []*v1.Pod{
+				st.MakePod().Namespace(ns).Name("t1-p1-1").Req(map[v1.ResourceName]string{v1.ResourceMemory: "50"}).Priority(
+					midPriority).Node(nodeName).Obj(),
+				st.MakePod().Namespace(ns).Name("t1-p1-2").Req(map[v1.ResourceName]string{v1.ResourceMemory: "50"}).Priority(
+					midPriority).Node(nodeName).Obj(),
+				st.MakePod().Namespace(ns).Name("t1-p1-3").Req(map[v1.ResourceName]string{v1.ResourceMemory: "50"}).Priority(
+					midPriority).Node(nodeName).Obj(),
+			},
+			status: []*v1alpha1.PodGroup{
```

Comment:
why this field is called status when it holds the list of PodGroups? Can we get rid of this and use only `podGroups` field?

Reply:
I apologize for the confusion caused by my naming.

The `status` field was actually used to check the intermediate status of `podGroup`s. I have now renamed the `status` and `want` fields to `intermediatePGState` and `expectedPGState`. The `intermediatePGState` field is necessary for us to check the changes in `podGroup`'s Status.
The purpose of this test is to ensure that the `podGroup` behave as expected as the `pods` change, so we need to test the intermediate state as well.


FYI, this is the sequence of steps in the integration test:
1. Create `PodGroup` (`podGroups`)
2. Create `Pod` (`existingPods`)
3. Check intermediate status of `PodGroup` (`intermediatePGStatus`)
4. Update `Pod` (`incomingPods`)
5. Check the final status of `PodGroup` (`expectedPGStatus`)

I referred to [elasticquota_controller_test.go](https://github.com/kubernetes-sigs/scheduler-plugins/blob/master/test/integration/elasticquota_controller_test.go) while working on this.

Reply:
Thanks for the explanation and rewording!